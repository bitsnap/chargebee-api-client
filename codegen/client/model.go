package client

import (
	"embed"
	"strings"

	"github.com/bitsnap/chargebee-api-client/codegen/scraper"
	. "github.com/bitsnap/go-util"
)

//go:embed model.go.tpl
var modelTemplateFS embed.FS

var modelTemplate = MustParseTemplate(modelTemplateFS, "model.go.tpl")

type Model struct {
	Name       string
	Attributes []*scraper.Attribute
}

func GenerateModel(name, header, page string) string {
	attributes, err := scraper.ScrapeAttributes(header, page)
	if err != nil {
		Logger().Fatalln(err)
	}

	output := strings.Builder{}
	err = modelTemplate.Execute(&output, Model{name, attributes})
	if err != nil {
		panic(err)
	}

	return output.String()
}
