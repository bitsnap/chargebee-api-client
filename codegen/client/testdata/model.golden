# Generated by goldga. DO NOT EDIT.
[snapshots]
"Model attributes code generation when generating models from scraped documentation pages should be able to generate models from templates" = '''
(string) (len=317) "type Entitlement struct {\n    Id string `json:\"id\" validate:\"required\"`\n    EntityId string `json:\"entity_id\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\"`\n    FeatureId string `json:\"feature_id\"`\n    FeatureName string `json:\"feature_name\"`\n    Value string `json:\"value\"`\n    Name string `json:\"name\"`\n}"
'''
